bemserver_ui/static/scripts/modules/components/charts/tsChartWeather.js

```diff
@@ -0,0 +1,338 @@
+    }
+
+    dispose() {
+        this.#chart.dispose();
+    }
+
+    showLoading() {
+        this.#chart.showLoading();
+    }
+
+    hideLoading() {
+        this.#chart.hideLoading();
+    }
+
+    load(name, dataset, timeFormat) {
+        console.log("Dataset",dataset);
+        this.hideLoading();
+
+        let options = this.#chart.getOption();
+        let listUnit = {0: [], 1: []};
+
+        options.legend[0].data = [];
+        options.legend[1].data = [];
+
+        options.title[0].text = `${name}`;
+        
+        options.toolbox[0].feature.dataView.optionToContent = (opt) => { return this.#optionToContent(opt, dataset, timeFormat); };
+
+        //options.tooltip[0].formatter = (params) => { return this.#tooltipFormatter(params, parameters.unit, timeFormat); };
+        
+        options.series.length = 0;
+
+        let series = [];
+        for (let [_parameter, serieParams] of Object.entries(dataset)) {
+            for (let [_settings, value] of Object.entries(serieParams)) {
+                let chartData = Object.entries(value.data).map(([date, value]) => [date, Parser.parseFloatOrDefault(value, Number.NaN, 2)]);
+                if (chartData.length > 0) {
+                    let serie = {
+                        name: value.name,
+                        type: "line",
+                        data: Object.entries(value.data).map(([date, value]) => [date, Parser.parseFloatOrDefault(value, Number.NaN, 2)]),
+                        emphasis: {
+                            focus: "series"
+                        },
+                        itemStyle: {
+                            color: this.#energyUseColors[value.name]
+                        },
+                        lineStyle: {
+                            width: 2,
+                            type: value.name.includes("forecast") ? "dashed" : "solid",
+                        },
+                        yAxisIndex: value.yAxis,
+                        unit: value.timeseries.unit,
+                    };
+                    series.push(serie);
+                    if (!listUnit[value.yAxis].includes(value.timeseries.unit))
+                        listUnit[value.yAxis].push(value.timeseries.unit);
+
+                    value.yAxis == 0 ? options.legend[0].data.push(value.name) : options.legend[1].data.push(value.name);
+                }
+            }
+        }
+        options.series = series;
+
+        options.yAxis[0].data = options.series.map((serie) => serie.name);
+        options.yAxis[0].axisLabel.formatter = (value, index) => { return `${value} (${listUnit[0]})`; };
+
+        options.yAxis[1].data = options.series.map((serie) => serie.name);
+        options.yAxis[1].axisLabel.formatter = (value, index) => { return `${value} (${listUnit[1]})`; };
+
+        console.log("Options",options.legend);
+
+        if (options.legend[1].data.length == 0) {
+            options.legend[0].left = "center";
+            options.legend[0].bottom = 10;
```

Comment:
Not sure that this bottom alignment is useful. I have the impression that it causes a misalignment of the legends from one chart to another. You should remove that.